lib/logqueue-fifo.c

```diff
@@ -267,6 +268,7 @@ log_queue_fifo_push_tail(LogQueue *s, LogMessage *msg, const LogPathOptions *pat
            * input thread finishes */
 
           main_loop_worker_register_batch_callback(&self->qoverflow_input[thread_id].cb);
+          self->qoverflow_input[thread_id].cb.user_data = log_queue_ref(s);
           self->qoverflow_input[thread_id].finish_cb_registered = TRUE;
```

Comment:
This ref will be called only at the first message of the job, not for all of the messages.
Add ref at the job starting and remove if the job is finished.